app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,39 @@
+class QueueItemsController < ApplicationController
+  before_filter :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    vid = Video.find(params[:video_id])
+    queue_video(vid)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if current_user_owner?(queue_item)
+    redirect_to my_queue_path
+  end
+
+  private
+
+  #create
+  def queue_video(video)
+    QueueItem.create(video: video, user: current_user, position:  new_queue_item_position) unless current_user_queued_video?(video)
+  end
+
+  def new_queue_item_position
+    current_user.queue_items.count + 1
+  end
+  
+  def current_user_queued_video?(video)
```

Comment:
This refactoring should definitely be changed as you are calling `current_user ...` this indicates that you want to have this method in User model.

HOMEWORK: refactor this to User model.